frontend/test/pytest/test_template.py

```diff
@@ -690,8 +1187,326 @@ def qnn(weights, inputs):
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_qrom(backend):
+    """Test QROM."""
+
+    def qrom():
+        qml.QROM(["1", "0", "0", "1"], control_wires=[0, 1], target_wires=[2], work_wires=[3])
+        return qml.probs(wires=3)
+
+    device = qml.device(backend, wires=4)
+    interpreted_fn = qml.QNode(qrom, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_prepselprep(backend):
+    """Test PrepSelPrep"""
+
+    params = np.array([0.4, 0.5, 0.1, 0.3])
+
+    def prepselprep(coeffs):
+        H = qml.ops.LinearCombination(
+            coeffs, [qml.Y(0), qml.Y(1) @ qml.Y(2), qml.X(0), qml.X(1) @ qml.X(2)]
+        )
+        qml.PrepSelPrep(H, control=(3, 4))
+        return qml.expval(qml.PauliZ(3) @ qml.PauliZ(4))
+
+    device = qml.device(backend, wires=5)
+    interpreted_fn = qml.QNode(prepselprep, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn(params)
+    jitted_result = jitted_fn(params)
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_mod_exp(backend):
+    """Test ModExp."""
+    x, b = 3, 1
+    base = 2
+    mod = 7
+
+    x_wires = [0, 1]
+    output_wires = [2, 3, 4]
+    work_wires = [5, 6, 7, 8, 9]
+
+    def mod_exp():
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.BasisEmbedding(b, wires=output_wires)
+        qml.ModExp(x_wires, output_wires, base, mod, work_wires)
+        return qml.sample(wires=output_wires)
+
+    device = qml.device(backend, wires=10, shots=2)
+    interpreted_fn = qml.QNode(mod_exp, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_multiplier(backend):
+    """Test Multiplier."""
+    x = 3
+    k = 4
+    mod = 7
+
+    x_wires = [0, 1, 2]
+    work_wires = [3, 4, 5, 6, 7]
+
+    def multiplier():
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.Multiplier(k, x_wires, mod, work_wires)
+        return qml.sample(wires=x_wires)
+
+    device = qml.device(backend, wires=8, shots=2)
+    interpreted_fn = qml.QNode(multiplier, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+@pytest.mark.xfail(reason="Multiple state preparations at beginning of circuit")
+def test_out_adder(backend):
+    """Test OutAdder."""
+    x = 5
+    y = 6
+    mod = 7
+
+    x_wires = [0, 1, 2]
+    y_wires = [3, 4, 5]
+    output_wires = [7, 8, 9]
+    work_wires = [6, 10]
+
+    def out_adder():
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.BasisEmbedding(y, wires=y_wires)
```

Comment:
BasisEmbedding are basically pauliX gates, you could put some X gates here and remove the xfail